### PR TITLE
Tweaked foam spread logic, foam layer.

### PR DESCRIPTION
**Pull request by xxalpha**
Foam now spreads under things even if they are dense.

foam1
foam2
foam3
foam4

### CODE REVIEW

**Comment by GunHog** [comment]
Makes sense :+1:

**Comment by Core0verload** [comment]
:+1:

**Comment by WJohn** [comment]
Why isn't it spreading under those chairs in those first two pictures?

**Comment by xxalpha** [comment]
First tile of foam fading.

**Comment by xxalpha** [comment]
Also I wouldn't say this is balance or fix. More of a feature.

**Comment by ExcessiveUseOfCobblestone** [comment]
Any Foamestry Buff is an A-Okay Buff :+1:

File: code/game/objects/effects/effect_system/effects_foam.dm [hunk 1]
```diff
@@ -96,13 +96,14 @@
...
 			continue
 
-		if(!T.Enter(src))
-			continue
-
 		var/obj/effect/effect/foam/foundfoam = locate() in T //Don't spread foam where there's already foam!
 		if(foundfoam)
 			continue
 
+		var/turf/t_loc = loc
```
Comment (RemieRichards) [inline code/game/objects/effects/effect_system/effects_foam.dm]: ```var/turf/t_loc = get_turf(src)```

File: code/game/objects/effects/effect_system/effects_foam.dm [hunk 2]
```diff
@@ -91,14 +91,8 @@
...
 /obj/effect/effect/foam/proc/spread_foam()
-	for(var/direction in cardinal)
-		var/turf/T = get_step(src,direction)
-		if(!T)
-			continue
-
-		if(!T.Enter(src))
-			continue
-
+	var/turf/t_loc = get_turf(loc)
```
Comment (RemieRichards) [inline code/game/objects/effects/effect_system/effects_foam.dm]: ```get_turf(src)```

File: code/game/objects/effects/effect_system/effects_foam.dm [hunk 3]
```diff
@@ -7,7 +7,7 @@
 	opacity = 0
 	anchored = 1
 	density = 0
-	layer = OBJ_LAYER + 0.05 //above table, below windoor/airlock/foamed metal.
+	layer = OBJ_LAYER - 0.5 //above table, below windoor/airlock/foamed metal.
```
Comment (phil235) [inline code/game/objects/effects/effect_system/effects_foam.dm]: @xxalpha why did you keep the now inaccurate comment?

**Comment by phil235** [comment]
This PR has reopened #5965

**Pull request merged**